crates/wasmparser/src/validator/types.rs

```diff
@@ -208,23 +205,35 @@ impl Type {
     /// Converts the type to a core function type.
     pub fn as_func_type(&self) -> Option<&FuncType> {
         match self {
-            Self::Func(ty) => Some(ty),
+            Type::Sub(SubType {
+                is_final: _,
+                supertype_idxs: _,
+                structural_type: StructuralType::Func(ft),
+            }) => Some(ft),
```

Comment:
You can use `..` to ignore multiple fields at a time, instead of writing them all out here and below:

```suggestion
            Type::Sub(SubType {
                structural_type: StructuralType::Func(ft),
                ..
            }) => Some(ft),
```